Recent commit with MDBX bindings broke windows builds

### Describe the bug

Originated from https://github.com/paradigmxyz/reth/pull/5941

Seems like some API is not exposed for windows

```
error[E0433]: failed to resolve: could not find `unix` in `os`
  --> crates\storage\db\src\implementation\mdbx\mod.rs:36:43
   |
36 |     Lazy::new(|| if cfg!(unix) { std::os::unix::process::parent_id() } else { std::process::id() });
   |                                           ^^^^ could not find `unix` in `os`

error[E0599]: no method named `set_handle_slow_readers` found for struct `EnvironmentBuilder` in the current scope
   --> crates\storage\db\src\implementation\mdbx\mod.rs:182:23
    |
182 |             inner_env.set_handle_slow_readers(
    |             ----------^^^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `set_max_readers`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `reth-db` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...


```

### Steps to reproduce

cargo build

### Node logs

_No response_

### Platform(s)

Windows (x86)

### What version/commit are you on?

latest

### What database version are you on?

_No response_

### What type of node are you running?

None

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

--- a/crates/storage/db/src/implementation/mdbx/mod.rs
+++ b/crates/storage/db/src/implementation/mdbx/mod.rs
@@ -32,8 +32,17 @@ const DEFAULT_MAX_READERS: u64 = 32_000;
 /// See [reth_libmdbx::EnvironmentBuilder::set_handle_slow_readers] for more information.
 const MAX_SAFE_READER_SPACE: usize = 10 * GIGABYTE;
 
-static PROCESS_ID: Lazy<u32> =
-    Lazy::new(|| if cfg!(unix) { std::os::unix::process::parent_id() } else { std::process::id() });
+static PROCESS_ID: Lazy<u32> = Lazy::new(|| {
+    #[cfg(unix)]
+    {
+        std::os::unix::process::parent_id()
+    }
+
+    #[cfg(not(unix))]
+    {
+        std::process::id()
+    }
+});
 
 /// Environment used when opening a MDBX environment. RO/RW.
 #[derive(Debug)]

